pkg/sql/exec_util.go

```diff
@@ -1057,6 +1057,17 @@ func golangFillQueryArguments(args ...interface{}) (tree.Datums, error) {
 				switch {
 				case val.IsNil():
 					d = tree.DNull
+				case val.Type().Elem().Kind() == reflect.String:
+					a := tree.NewDArray(types.String)
+					// TODO(angelaw): Ranging through val seems to require pointer arithmetic so did it this way
+					for s := range args {
```

Comment:
I think this loop needs to look more like this:

```
					for i := 0; i < val.Len(); i += 1 {
						a.Append(tree.NewDString(val.Index(i).String()))
					}
			
```

dealing with reflected types is weird.. but the idea is that instead of using the built in slice operations like `len()` or the `[]` index access, there are separate functions for that

also note that we don't want to loop over `args` since the array is just one of the arguments to `golangFillQueryArguments`

Reply:
to be safe, we also should check if the result of `val.Index(i)` is `nil`, and if so we'd want to append `tree.DNull` in the array instead of a `DString`

Reply:
`val.Index(i)` is a string which can't be `nil` in Go I believe — I added a check for `isValid` but I'm not sure it's necessary at all.